components/src/modal/useFocus.ts

```diff
@@ -0,0 +1,59 @@
+import { useRef, useMemo } from 'react';
+
+function useFocusTrap(open: boolean) {
+  const firstRef = useRef<HTMLElement | null>(null);
+  const lastRef = useRef<HTMLElement | null>(null);
+
+  const findFocusable = (
+    modal: HTMLElement,
+    focusableList: HTMLElement[] = []
+  ) => {
+    for (const child of Array.from(modal.children) as HTMLElement[]) {
+      if (child.tabIndex >= 0) {
+        focusableList.push(child);
+      }
+
+      findFocusable(child, focusableList);
+    }
+
+    return focusableList;
+  };
+
+  const setRef = (modal: HTMLDialogElement) => {
+    if (!modal || !open) {
+      return;
+    }
+
+    const focusableList = findFocusable(modal);
+    if (focusableList.length === 0) {
+      return;
+    }
+
+    firstRef.current = focusableList[0];
+    lastRef.current = focusableList[focusableList.length - 1];
+
+    firstRef.current.focus();
+  };
+
+  const handleKeyDownTab = (event: React.KeyboardEvent<HTMLDialogElement>) => {
+    if (event.key !== 'Tab') {
+      return;
+    }
+
+    if (event.shiftKey) {
+      if (document.activeElement === firstRef.current) {
+        event.preventDefault();
+        lastRef.current?.focus();
+      }
+    } else {
+      if (document.activeElement === lastRef.current) {
```

Comment:
아래처럼 작성하지 않은 이유가 있을까요?

```suggestion
    } else if (document.activeElement === lastRef.current) {
```

Reply:
엇 `else if`의 존재를 잊고 있었습니다. 😅 아래 코멘트의 내용과 종합하여, 가독성을 개선할 수 있도록 수정해보겠습니다!

Reply:
if문 조건을 논리연산자로 조합하여, if-else문을 제거했습니다!

[refactor: 불필요한 if-else문 제거 (b9db0bd)](https://github.com/woowacourse/react-modules/pull/157/commits/b9db0bd309d816a344bf6d22f2ac0db6fddf971b)